File: 13/list-13.go
package main

import "fmt"

type Matematica struct{}

func (m Matematica) Fatorial(n int) int {
	if n < 0 {
		fmt.Println("Fatorial não existe para números negativos.")
		return -1
	}
	if n == 0 || n == 1 {
		return 1
	}
	return n * m.Fatorial(n-1)
}

func main() {
	m := Matematica{}
	numero := 5
	fmt.Printf("O fatorial de %d é %d\n", numero, m.Fatorial(numero))
}

File: 13/list-13.py
class Matematica:
    @staticmethod
    def fatorial(n):
        if n < 0:
            raise ValueError("Negativos não têm fatorial.")
        return 1 if n in (0, 1) else n * Matematica.fatorial(n - 1)

def main():
    numero = int(input('digite o numero: '))
    fatorial = Matematica.fatorial(numero)
    print(f"O fatorial de {numero} é {fatorial}")

if __name__ == "__main__":
    main()
